yjit.c

```diff
@@ -1116,6 +1116,17 @@ rb_yjit_assert_holding_vm_lock(void)
     ASSERT_vm_locking();
 }
 
+// If true, count the number of insns executed on vm_exec_core.
+// Using a C variable instead of Rust code to minimize the overhead.
+bool rb_yjit_insn_counter_enabled = false;
+
+// Enable counting the number of insns executed on vm_exec_core.
+void
+rb_yjit_insn_counter_enable(void)
+{
+    rb_yjit_insn_counter_enabled = true;
+}
```

Comment:
Maybe we could get rid of this state if we always do the update/increment?

Reply:
Incrementing a register is fast, but this flag is for propagating a value from a register to Rust. The latter is a function call, so I'd like to avoid it by just checking a C variable.

Reply:
Hm, I guess you're thinking about moving ownership to C. Maybe that works? I'll try.

Reply:
removed this function.